### DISCUSSION TITLE
Getting App store signedTransactionInfo decoded

### DISCUSSION THREAD

**Discussion by kingzee100** [imdhemy/laravel-in-app-purchases#333] [Q&A]
Hi, when getting a server notification from the Apple app store: in the listener file the $subscription->getExpiryTime() returns an object:
example:

Imdhemy\AppStore\ValueObjects\Time Object
(
    [timestampMilliseconds:Imdhemy\AppStore\ValueObjects\Time:private] => 1683196418000
)

as you can see the value is protected by a private method. How can I get the value: 1683196418000?

**Comment by imdhemy** [comment]
You have two options:

1. From the Datetime

```php
$timestamp = $expiryTime->toDateTime()->getTimestamp();
```

2. From Carbon

```php
$timestamp = $expiryTime->toCarbon()->getTimestamp();
```

You still have access to the original milliseconds from the Carbon instance
```php
$milliseconds = $expiryTime->toCarbon()->getTimestampMs();
```

**Comment by kingzee100** [comment]
Works great. Thanks!

**Comment by kingzee100** [comment]
Hi, thanks for making this package! It's certainly a big help.

I have everything working, including receipt validation and server
notifications for both Apple and Google :)

On android, however, when I test (sandbox), I only get server
notification for the initial Subscription event and the Expiration event
(about 5 minutes later) but not for the renewals.

I did define the event SubscriptionRenewed in the Iiap.config in the same
way as I did for all other events (all work).

but for some reason the renewal events do not seem to get called.

Any ideas on what could be the reason?

Thanks again,

Zak

**Reply by imdhemy** [reply]
Hi Zak,

Happy to hear that this package helped you. I believe this is related to Google or console configurations. I bet you already had a look on this document as well, didn't you? https://developer.android.com/google/play/billing/test

**Comment by kingzee100** [comment]
Yes it was. Thanks so much for the reply. I appreciate it.

Great package!
